targets/TARGET_RENESAS/TARGET_RZ_A1XX/TARGET_GR_LYCHEE/trng_api_esp32.cpp

```diff
@@ -59,19 +64,24 @@ extern "C" int trng_get_bytes_esp32(uint8_t *output, size_t length, size_t *outp
     char recv_data[4];
     size_t idx = 0;
     int i;
-    int err_cnt = 0;
+    int retry_cnt = 0;
 
     while (idx < length) {
         send_data[0] = 0;
         ret = mI2c.write(ESP32_I2C_ADDR, send_data, 1);
         if (ret == 0) {
-            mI2c.read(ESP32_I2C_ADDR, recv_data, sizeof(recv_data));
-            for (i = 0; (i < 4) && (idx < length); i++) {
+            ret = mI2c.read(ESP32_I2C_ADDR, recv_data, sizeof(recv_data));
+            if (ret != 0) {
+                idx = 0;
+                break;
```

Comment:
No retry in case of read error. How is it different then a write error?

Reply:
@mazimkhan There has already been some discussion on this. See above.